hpy/devel/include/hpy/hpymodule.h

```diff
@@ -19,9 +19,18 @@ extern HPyContext *_ctx_for_trampolines;
 #endif
 
 
+/**
+ * Unlike CPython, HPy module definition does not specify module name.
+ * The name if always taken from the ModuleSpec (PEP 451).
+ */
 typedef struct {
-    const char* name;
     const char* doc;
+    /**
+     * Size of the module state structure. If set to zero, then the module
+     * will not get allocated and assigned any HPy module state. Negative
+     * size, unlike in CPython API, does not have any specific meaning and
+     * will produce a runtime error.
```

Comment:
Ahh, this is the answer to my question above. Is this rendered somewhere in the html documentation?

Reply:
Yes that's good point. I did cursory review of the docs and fixed all examples that are compiled/tested, but I should do more thorough pass over it! However, I suggest I do that on top of @fangerer current documentation improvements PR (and merge this PR as-is w.r.t. documentation) to reduce conflicts if that sounds good?

Reply:
Sounds good, thanks